docs/custom-plugins/custom-rules.md

```diff
@@ -0,0 +1,117 @@
+# Rules in plugins
+
+Rules are used to make sure that the API description is in the expected format and aligns with the expected API standards. Before you build any custom rules:
+
+* Learn more about [Redocly rules](../rules.md).
+* Check the list of [built-in rules](../rules/built-in-rules.md).
+* See if you can build a [configurable rule](../rules/configurable-rules.md) to meet your needs.
+
+Exhaust the above options first, because they are simpler and more maintainable than building a custom plugin. If you need to build your own rules though, then you're in the right place! Read on ...
+
+## Build the custom rule
+
+Each rule is a function that returns an object with methods that Redocly CLI calls to "visit" nodes while traversing the definition document. The object keys are the node types that are encountered in the document. In this simple example, the custom plugin holds a rule that fails if any `operationId` is set to "test".
+
+To keep the plugin code manageable, each rule can go in its own file. This example is in `plugins/rules/opid-not-test.js`:
+
+```js
+module.exports = OperationIdNotTest;
+
+function OperationIdNotTest() {
+  return {
+    Operation(operation, ctx) {
```

Comment:
One more place with implicit usage of  the visitor func.

Reply:
Thanks! Fixed it :)